### PR TITLE
Bug fix/gal/39 fix api

### PR DESCRIPTION
**Pull request by galdd**

### CODE REVIEW

File: src/@types/expenseListsResponse [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { ExpenseLIstType } from "./expense-list-prop";
+import { SortOrder } from "./sortOrderTypes";
+
+export interface ExpenseListsResponse {
+  data: ExpenseLIstType[];
+  isLoading: boolean;
+  error: Error | null;
```
Comment (shakedlv) [inline src/@types/expenseListsResponse:7]: use error?: Error instead of Error | Null

**Comment by galdd** [comment]
https://github.com/ofir-zeitoun/expenses-client/assets/11245756/596c13f5-ea72-41ad-828b-e77056e5595c

File: src/@types/expenseListsResponse (hunk 1 above)
Comment (galdd) [reply src/@types/expenseListsResponse:7]: fixed

File: src/components/dashboard/expense-list-section/expense-list-section.tsx [hunk 2]
```diff
@@ -1,29 +1,72 @@
...
+    if (section?.data) {
+      if (offset === 0) {
+        setLists(section.data);
+      } else {
+        setLists((prevLists) => [...prevLists, ...section.data]);
+      }
+    }
+  }, [section, offset]);
+
+  useEffect(() => {
```
Comment (shakedlv) [inline src/components/dashboard/expense-list-section/expense-list-section.tsx:31]: Ofir : Please explain this is not best practice

File: src/components/dashboard/expense-list-section/expense-list-section.css [hunk 3]
```diff
@@ -5,4 +5,19 @@
     flex-direction: column;
     gap: 1.875rem;
     padding: 0 2.5rem 0.625rem 0.25rem;
+}
+
+.load-more-button {
+    padding: 0.75rem 1.5rem;
+    background-color: #007bff;
```
[the remark this answers is not in the document]
Comment (galdd) [reply src/components/dashboard/expense-list-section/expense-list-section.css:12]: fixex, and I will change all the elements in the projects to dark/light mode in other branch

File: src/components/dashboard/expense-list-section/expense-list-section.tsx [hunk 4]
```diff
@@ -1,29 +1,72 @@
...
+  const [lists, setLists] = useState<ExpenseLIstType[]>([]);
+  const {
+    data: section,
+    isLoading,
+    error,
+  } = useExpenseLists(offset, 5, sortOrder);
+
+  const updateLists = useCallback(() => {
+    if (section?.data) {
+      if (offset === 0) {
```
[the remark this answers is not in the document]
Comment (galdd) [reply src/components/dashboard/expense-list-section/expense-list-section.tsx:23]: fixed

File: src/components/dashboard/expense-list-section/expense-list-section.tsx (hunk 2 above)
Comment (ofir-zeitoun) [reply src/components/dashboard/expense-list-section/expense-list-section.tsx:28]: you invoke it when it changes, not when you decide

File: src/components/dashboard/expense-list-section/expense-list-section.tsx [hunk 5]
```diff
@@ -1,29 +1,69 @@
...
+  const {
+    data: section,
+    isLoading,
+    error,
+  } = useExpenseLists(offset, 5, sortOrder);
+
+  const updateLists = useCallback(() => {
+    if (section?.data) {
+      setLists((prevLists) =>
+        offset === 0 ? section.data : [...prevLists, ...section.data]
```
Comment (ofir-zeitoun) [inline src/components/dashboard/expense-list-section/expense-list-section.tsx:23]: same solution as before, no need to branch.
if you do ```setList(prev => [...prev, section.data])``` you'll have the correct behavior.
when offset is 0, you set list to [].

File: src/components/dashboard/expense-list-section/expense-list/expense-list.tsx [hunk 6]
```diff
@@ -1,22 +1,43 @@
...
+import AddExpenseItem from "./add-expense-item/add-expense-item";
+
+export const ExpenseList = React.memo(({ list }: { list: ExpenseLIstType }) => {
+  const [sortOrder, setSortOrder] = useState<"asc" | "desc">("asc");
+
+  const toggleSortOrder = () => {
+    setSortOrder(sortOrder === "asc" ? "desc" : "asc");
+  };
+
+  const sortedExpenses = [...list.expenses].sort((a, b) => {
```
Comment (ofir-zeitoun) [inline src/components/dashboard/expense-list-section/expense-list/expense-list.tsx:17]: is toSorted workes?

File: src/components/dashboard/expense-list-section/expense-list/expense-list.tsx [hunk 7]
```diff
@@ -1,22 +1,43 @@
...
+    if (sortOrder === "asc") {
```
Comment (ofir-zeitoun) [inline src/components/dashboard/expense-list-section/expense-list/expense-list.tsx:18]:
```const multiply = {
  asc: 1,
  desc: -1
}
return (a.price - b.price) * multiply[sorOrder]
```

**Review by ofir-zeitoun — APPROVED** [review]

**Pull request merged**